platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/LongListOffHeap.java

```diff
@@ -169,7 +169,8 @@ protected void writeLongsData(final FileChannel fc) throws IOException {
             for (int i = firstChunkWithDataIndex; i < totalNumOfChunks; i++) {
                 final ByteBuffer byteBuffer = chunkList.get(i);
                 final ByteBuffer nonNullBuffer = requireNonNullElse(byteBuffer, emptyBuffer);
-                final ByteBuffer buf = nonNullBuffer.slice(); // slice so we don't mess with state
+                final ByteBuffer buf =
+                        nonNullBuffer.slice(0, nonNullBuffer.capacity()); // slice so we don't mess with state
```

Comment:
Not sure the comment is accurate.  The buffer returned from slice is still backed by the same internal data as the original (so writes to one are reflected in the other).

> Changes to this buffer's content will be visible in the new buffer, and vice versa

Reply:
Fair enough. I updated the comment.